src/FlightMap/Widgets/VehicleWarningSideBar.qml

```diff
@@ -0,0 +1,128 @@
+
+import QtQuick          2.12
+import QtQuick.Layouts  1.12
+import QtQuick.Controls 2.4
+
+import QGroundControl               1.0
+import QGroundControl.Controls      1.0
+import QGroundControl.ScreenTools   1.0
+import QGroundControl.Palette       1.0
+
+
+Rectangle {
+    id:      vehicleWarningSideBarRect
+    color:   qgcPal.window
+    radius:  _toolsMargin / 2
+    opacity: 0.8
+
+    // Overridden by parent
+    property int maxWidth:              400
+    property int minWidth:              200
+    property int instrumentPanelHeight: 100
+    property int winchWidgetHeight:     200
+
+    property ListModel errorListModel:  ListModel {}
+    property var _activeVehicle:        QGroundControl.multiVehicleManager.activeVehicle
+    property int maxHeight:             parent.height * 0.5
+    property int maxWidthBasedOnWindow: parent.width * 0.25
+    property int minTopMargin:          instrumentPanelHeight + ScreenTools.defaultFontPixelHeight / 2
+    property int minBottomMargin:       winchWidgetHeight + ScreenTools.defaultFontPixelHeight / 2
+
+    anchors {
+        margins:      -ScreenTools.defaultFontPixelHeight
+        bottom:       parent.bottom
+        bottomMargin: Math.max(parent.height * 0.3, minBottomMargin)
+    }
+    
+    height:  Math.min(Math.min(vehicleWarningSideBarLayout.implicitHeight + (ScreenTools.defaultFontPixelWidth * 4), maxHeight), parent.height - minTopMargin - minBottomMargin)
+    width:   Math.min(Math.max(maxWidthBasedOnWindow, minWidth), maxWidth)
+    visible: _activeVehicle && errorListModel.count > 0
+
+    ColumnLayout {
+        id:              vehicleWarningSideBarLayout
+        spacing:         _toolsMargin * 2
+        anchors.fill:    parent
+        anchors.margins: ScreenTools.defaultFontPixelWidth * 2
+
+        RowLayout {
+            Layout.fillWidth: true
+
+            QGCLabel {
+                text:             errorListModel.count > 0 ? "Warnings" : "No Warnings"
```

Comment:
Ah I see, would "Critical messages" work? Or removing the label and just keep the trashcan-icon at the top of the container. I think it could look a bit off to move it out of the container

Reply:
I went with removing the label, but keeping the trashcan icon inside the container, can change later if we think its necessary